articles/why-and-how-zk-snark-works/3_non_interactive_zero-knowledge_of_a_polynomial.md

```diff
@@ -0,0 +1,260 @@
+---
+Title: Non interactive Zero-knowledge of a polynomial
+Status: Done
+Level: "3"
+---
+# Chapter 3 - Non-Interactive Zero Knowledge of a Polynomial
+
+## 3.1. Proving Knowledge of a Polynomial
+
+We start with a problem of proving the knowledge of a polynomial and make our way to a generic approach. We will discover many other
+properties of polynomials along the way.
+
+The discussion so far has focused on a weak notion of a proof, where parties have to trust each other because there are no measures yet
+to enforce the rules of the protocol. For example, the prover is not required to know a polynomial, and he can use any other means
+available to him to come up with a correct result. Moreover, if the amplitude of the verifier’s polynomial evaluations is not large ,
+let us say 10, the verifier can guess a number, and there is a non-negligible probability that it will be accepted. We have to address
+such weakness of the protocol, but first what does it means to know a polynomial? A polynomial can be expressed in the form :
+$$c_{n}x^{n}+ \cdots + c_{1}x^{1}+c_{0}x^{0} $$
+If one stated that he or she knows a polynomial of degree 1
+$(i.e., c_{1}x^{1}+c_{0})$, that means that what one really knows is the coefficients $c_{0}, c_{1}$. Moreover, coefficients can have
+any value, including 0. Let us first look more closely at the anatomy of the solution.
+
+## 3.2. **Factorization**
+
+It states that any polynomial can be factored into linear polynomials as long as it is solvable.
```

Comment:
Link to the appropriate docs instead of rewriting them. Write the example in the origin docs if necessary

Reply:
Check the rest of this doc